tiny_dnn/core/framework/tensor.h

```diff
@@ -41,6 +41,14 @@ class Tensor {
   Tensor() {}
 
   /**
+   * Constructor that accepts a storage object and create a Tensor housing
+   * that storage. Commonly used by [] operator.
+   * @param storage object, can be xtarray.
+   * @return
+   */
+  explicit Tensor(Storage s) : storage_(s) {}
```

Comment:
@beru yeah, lot of literature about this but the more context the better. At least is what we teach to the guys at university. Agree, @karandesai-96 pass storage by reference.